controllers/vspherecluster_controller.go

```diff
@@ -240,6 +243,29 @@ func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconci
 	}
 	conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
 
+	// Remove finalizer on Identity Secret
+	if identity.IsSecretIdentity(ctx.VSphereCluster) {
+		secret := &apiv1.Secret{}
+		secretKey := client.ObjectKey{
+			Namespace: ctx.VSphereCluster.Namespace,
+			Name:      ctx.VSphereCluster.Spec.IdentityRef.Name,
+		}
+		err := ctx.Client.Get(ctx, secretKey, secret)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		if !apierrors.IsNotFound(err) {
```

Comment:
remove this If, we want this to happen anyway, as it'd be the case where the secret was found